juju1/state/migration_export.go

```diff
@@ -637,17 +642,34 @@ func (e *exporter) newAddressArgs(a address) description.AddressArgs {
 	}
 }
 
-func lxcToLXDInstance(id string) string {
+func lxcToLXDInstance(modelUUID, id string) (string, error) {
 	parts := strings.Split(id, "-")
-	if len(parts) == 5 {
-		parts[3] = lxcToLXD(parts[3])
+	if len(parts) != 5 || parts[0] != "juju" || parts[1] != "machine" {
+		// Not a container, leave it.
+		return id, nil
+	}
+	parts[3] = lxcToLXD(parts[3])
```

Comment:
Is parts guaranteed to have at least 4 elements?

Reply:
Yup - there's an early return above if it doesn't have 5 elements.

Reply:
ah right, sorry.